LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,496 @@
+                Console.WriteLine(ex.Message);
+                Environment.Exit(1);
+            }
+        }
+
+        public async Task CheckAndUploadLinuxContainerImageAsync(string imageSkipToken = null, int pageSize = 10)
+        {
+            do
+            {
+                var ContainerImagesRequest = new ListContainerImagesRequest
+                {
+                    SkipToken = imageSkipToken,
+                    PageSize = pageSize
+                };
+                var ContainerImagesResponse = await PlayFabMultiplayerAPI.ListContainerImagesAsync(ContainerImagesRequest);
+
+                ErrorCheck(ContainerImagesResponse.Error);
+
+                imageSkipToken = ContainerImagesResponse.Result.SkipToken ?? null;
+                IEnumerable<string> existingImageNames = ContainerImagesResponse.Result.Images.Where(x => x == settings.ContainerStartParameters.ImageDetails.ImageName);
+
+                if (!existingImageNames.Any() && string.IsNullOrEmpty(imageSkipToken))
+                {
+                    await UploadLinuxContainerImageAsync(settings.ContainerStartParameters.ImageDetails);
+                }
+                else if (existingImageNames.Any())
+                {
+                    var ContainerImageTagsRequest = new ListContainerImageTagsRequest
+                    {
+                        ImageName = existingImageNames.First() ?? null,
+                    };
+                    var ContainerImageTagsResponse = await PlayFabMultiplayerAPI.ListContainerImageTagsAsync(ContainerImageTagsRequest);
+
+                    ErrorCheck(ContainerImageTagsResponse.Error);
+
+                    IEnumerable<string> existingImageTags = ContainerImageTagsResponse.Result.Tags.Where(x => x == settings.ContainerStartParameters.ImageDetails.ImageTag);
+
+                    if (existingImageTags.Any())
+                    {
+                        Console.WriteLine("Your container image is already uploaded. Continuing deployment...");
+                        break;
+                    }
+
+                    if (!existingImageTags.Any() && string.IsNullOrEmpty(imageSkipToken))
+                    {
+                        await UploadLinuxContainerImageAsync(settings.ContainerStartParameters.ImageDetails);
+                    }
+                }
+
+            } while (!string.IsNullOrEmpty(imageSkipToken));
+        }
+
+        public async Task<PlayFabResult<EmptyResponse>> UploadCertificateAsync(GameCertificateDetails certificate)
+        {
+            X509Certificate2 certCopy = new X509Certificate2(certificate.Path);
+            var uploadCertificateRequest = new UploadCertificateRequest
+            {
+                GameCertificate = new Certificate
+                {
+                    Name = certificate.Name,
+                    Base64EncodedValue = Convert.ToBase64String(certCopy.RawData)
+                    //Password = ""  //Only passwordless certificates supported currently
```

Comment:
is this a TODO? If it is, we should create an issue on the repository and link it here

Reply:
Not necessarily a TODO, just a note to people (devs) so they know that `Password` is a possible parameter but we currently do not support that. I guess we could make it a TODO if we have customers request?
Is this confusing?

Reply:
people that download LocalMultiplayerAgent via the "Releases" tab won't see this comment, since they download the code already compiled. The only way they can see it is if they check the LMA source code (in most cases they don't) or if they see an issue on the repository that this is a "known issue"

Reply:
@emmanuel-jr  make sure to add this to the readme